### PR TITLE
feat(dev-catalog): Rename & Reorder existing filters in Dev Catalog

### PR DESCRIPTION
**Pull request by abhinandan13jan**
Addresses - https://issues.redhat.com/browse/ODC-2454

 Renames & Reorders existing filters in Dev Catalog
P.S: Helm is not added as it will be added in a later PR https://github.com/openshift/console/pull/3850
Screencast from 01-10-2020 04_41_17 PM

### CODE REVIEW

**Comment by rohitkrai03** [comment]
@abhinandan13jan Your PR seems to have broken the logic for the badges in each card. It was working previously and is there in the design docs of your story as well. But you screencast doesn't have that.

File: frontend/public/components/catalog/catalog-items.jsx [hunk 1]
```diff
@@ -98,38 +98,74 @@ const pageDescription =
   'Catalog. Cluster admins can install additional apps which will show up here automatically.';
 
 // Filter property white list
-const filterGroups = ['kind'];
+const filterGroups = ['type'];
```
Comment (rohitkrai03) [inline frontend/public/components/catalog/catalog-items.jsx]: Why are you changing `kind` -> `type`?

File: frontend/public/components/catalog/catalog-items.jsx [hunk 2]
```diff
@@ -98,38 +98,74 @@ const pageDescription =
...
 
 const getAvailableFilters = (initialFilters) => {
   const filters = _.cloneDeep(initialFilters);
-  filters.kind = {
-    ClusterServiceClass: {
-      label: 'Service Class',
-      value: 'ClusterServiceClass',
+  filters.type = {
```
Comment (rohitkrai03) [inline frontend/public/components/catalog/catalog-items.jsx]: I do not get this change? Why was this necessary? The filters worked out previously as intended even without these changes.

File: frontend/public/components/catalog/catalog-items.jsx [hunk 3]
```diff
@@ -98,38 +98,74 @@ const pageDescription =
...
+      active: true,
+      kinds: {
+        ClusterServiceVersion: {
+          label: 'Installed Operators',
+          value: 'InstalledOperator',
+          active: false,
+        },
+      },
+    },
+    helm: {
```
Comment (rohitkrai03) [inline frontend/public/components/catalog/catalog-items.jsx]: Do not add the helm filter here as there is no helm integration till now. My PR adds the filter for that - https://github.com/openshift/console/pull/3850

File: frontend/public/components/catalog/catalog-items.jsx [hunk 4]
```diff
@@ -98,38 +98,74 @@ const pageDescription =
...
+      label: 'Helm Charts',
       active: false,
+      kinds: {
+        ImageStream: {
```
Comment (rohitkrai03) [inline frontend/public/components/catalog/catalog-items.jsx]: This is wrong because Helm charts are not based on `ImageStreams`

File: frontend/public/components/catalog/catalog-items.jsx (hunk 2 above)
Comment (rohitkrai03) [reply frontend/public/components/catalog/catalog-items.jsx]: The changes here also broke the badge feature. Now there are no badges being shown on the card.

File: frontend/public/components/catalog/catalog-items.jsx (hunk 1 above)
Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-items.jsx]: filterGroup has now been changed to `type` instead of `kind`. The ability of a `type` is to hold more than one `kind` wherever required.

File: frontend/public/components/catalog/catalog-items.jsx (hunk 2 above)
Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-items.jsx]: filterGroup has now been changed to `type` instead of `kind`. The ability of a `type` is to hold more than one `kind` wherever required.

Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-items.jsx]: The badges work perfectly with the latest changes

Comment (rohitkrai03) [reply frontend/public/components/catalog/catalog-items.jsx]: I don't see any badges in the screencast that you've shared with the PR.

Comment (rohitkrai03) [reply frontend/public/components/catalog/catalog-items.jsx]: Do you have an example type right now that has more kinds in it?

File: frontend/public/components/catalog/catalog-items.jsx (hunk 3 above)
Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-items.jsx]: I could remove this. But the filters in the referenced PR wouldn't work with type integration.

Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-items.jsx]: This works perfectly with the existing item structure. Hence, if the items with kind `HelmChart` are added, it will fit in without the need to make any changes in the filters.

File: frontend/public/components/catalog/catalog-items.jsx (hunk 4 above)
Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-items.jsx]: forgot to update the name

File: frontend/public/components/catalog/catalog-items.jsx (hunk 3 above)
Comment (rohitkrai03) [reply frontend/public/components/catalog/catalog-items.jsx]: I don't see a point adding a filter that doesn't have any item in it.

File: frontend/public/components/catalog/catalog-items.jsx (hunk 1 above)
Comment (rohitkrai03) [reply frontend/public/components/catalog/catalog-items.jsx]: Do you have a use case that fits this type of change? The current acceptance criteria for the story doesn't seem like it needs this change at the moment.

File: frontend/public/components/catalog/catalog-items.jsx (hunk 2 above)
Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-items.jsx]: No, but the initial plan was to put in unmanged services as a `type` with `imageStream`, `templates` & `ServiceClass`. I wouldn't see a change from the filterGroup `kind` if there wasn't a need for a type as such :)

Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-items.jsx]: Updated the mockups. The badges are there.

File: frontend/public/components/catalog/catalog-items.jsx (hunk 3 above)
Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-items.jsx]: Just part of the UX for accommodating future needs, IMO.

File: frontend/public/components/catalog/catalog-items.jsx (hunk 2 above)
Comment (rohitkrai03) [reply frontend/public/components/catalog/catalog-items.jsx]: But i don't see a change where umanaged services are grouped with kinds `imageStream`, `template` and `serviceClass`. They are all a separate type right now.

Comment (rohitkrai03) [reply frontend/public/components/catalog/catalog-items.jsx]: The way I see it right now is that all this change is unnecessary. The same could be achieved just by changing the name `Installed operators` -> `Operator Backed`, changing the order of the filters and just default the `active: true` for operator backed filter.

Comment (rohitkrai03) [reply frontend/public/components/catalog/catalog-items.jsx]: Plus, i don't see the point of grouping multiple `kinds` into a `type` for filters. In the end it just needs to filter the items in catalog with ability to have multiple filters selected at once. That was already working perfectly fine whichout these changes. Am not sure if am missing something else here. That's why i added comments to the story as well.

Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-items.jsx]: that is true. But this adheres to the initial discussion regarding the implementation. The changes can be reverted, if UX confirms that there is no need for grouping of the `kinds` in the near future.  Anyway I'll make the changes go away in that case.

File: frontend/public/components/catalog/catalog-items.jsx (hunk 3 above)
Comment (rohitkrai03) [reply frontend/public/components/catalog/catalog-items.jsx]: Well, I think those future needs should be handled by the future story that adds that feature in future. That is actually part of the acceptance criteria for the Helm stories and not part of this one. Even thought the mockups show that in this one as well. We can clarify this with UX. 
cc: @serenamarie125

File: frontend/public/components/catalog/catalog-items.jsx (hunk 2 above)
Comment (rohitkrai03) [reply frontend/public/components/catalog/catalog-items.jsx]: I just want to understand from an implementation point of view what benefits we are getting by grouping kinds here. UX may be thinking it in a way that all unmanaged service are in one group but purely from implementation and code changes what will be the benefits?

Comment (rohitkrai03) [reply frontend/public/components/catalog/catalog-items.jsx]: From a flat data structure you're making it into a nested object struture. And then making the changes to the filtering logic to accomodate the change in data structure. Plus there is no actual grouping of `kinds` happening here. That doesn't seem right to me.

**Comment by abhinandan13jan** [comment]
@rohitkrai03 Please re-review, I have removed all grouping logics. The PR currently Renames & Reorders existing filters in Dev Catalog

File: frontend/public/components/catalog/catalog-items.jsx (hunk 3 above)
Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-items.jsx]: I've put out Helm from the filters

File: frontend/public/components/catalog/catalog-items.jsx (hunk 4 above)
Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-items.jsx]: Removed all kind groupings

File: frontend/public/components/catalog/catalog-items.jsx (hunk 2 above)
Comment (abhinandan13jan) [reply frontend/public/components/catalog/catalog-items.jsx]: The existing data structure used for the filters is neither a flat, nor a linear one. It comprises a single layer nesting of objects by `filterGroups`. For adding flexibility of having multiple kinds per `key` in a `filterGroup`, I simply added another layer of objects by nesting the same existing structure of data, which I referred to as grouping. However, this has been reverted.

**Comment by christianvogt** [comment]
The patternfly catalog, along with operator hub use title case for the `Type`. I understand that the mock ups show `TYPE` with a different style. But to stay consistent with Operator Hub I suggest we keep it as `Type` for now.

**Comment by andrewballantyne** [comment]
> From a UX perspective, the TYPE filters and card badges look as designed. Nice work!

> The patternfly catalog, along with operator hub use title case for the Type. I understand that the mock ups show TYPE with a different style. But to stay consistent with Operator Hub I suggest we keep it as Type for now.

@christianvogt and @serenamarie125 I think you're in conflict with your statements.

**Comment by serenamarie125** [comment]
@christianvogt I didn't see that comment - the style of TYPE should definitely match what it is on OperatorHub, although I'm not sure you changed that

File: frontend/public/components/catalog/catalog-items.jsx [hunk 5]
```diff
@@ -129,7 +129,7 @@ const getAvailableFilters = (initialFilters) => {
 };
 
 const filterGroupNameMap = {
-  kind: 'Type',
+  kind: 'TYPE',
```
Comment (andrewballantyne) [inline frontend/public/components/catalog/catalog-items.jsx]: @serenamarie125 this change makes the two pages unbalanced is what @christianvogt is saying.

Today (before this change) this is the two views:

Screen Shot 2020-01-10 at 3 18 33 PM
Screen Shot 2020-01-10 at 3 18 58 PM

So changing it to all capitals makes it not match the Operator Hub (and the colour / spacing is different than the designs)

**Comment by abhinandan13jan** [comment]
@christianvogt changed TYPE -> Type, 
@andrewballantyne Fixed filters
